lib/plugins/screenshot/index.js

```diff
@@ -1,46 +1,98 @@
 'use strict';
 
 const Promise = require('bluebird');
+const sharp = require('sharp');
+const get = require('lodash.get');
 
-function getImagesAndName(images) {
+const defaultConfig = {
+  type: 'png',
+  png: {
+    compressionLevel: 6
+  },
+  jpg: {
+    quality: 80
+  },
+  maxSize: 2000
+};
+
+function getImagesAndName(images, imageType) {
   return images.map((image, index) => {
     return {
       data: image,
-      name: index + '.png'
+      name: index + '.' + imageType
     };
   });
 }
 
-function storeScreenshots(url, imagesAndName, storageManager) {
-  return Promise.map(imagesAndName, screenshot =>
-    storageManager.writeDataForUrl(
-      screenshot.data,
-      screenshot.name,
-      url,
-      'screenshots'
-    )
-  );
+function storeScreenshots(url, imagesAndName, storageManager, options) {
+  return Promise.map(imagesAndName, screenshot => {
+    const maxSize = get(options, 'maxSize', defaultConfig.maxSize);
+    const imageType = get(options, 'type', defaultConfig.type);
+    if (imageType === 'png') {
+      const compressionLevel = get(
+        options,
+        'png.compressionLevel',
+        defaultConfig.png.compressionLevel
+      );
+      sharp(screenshot.data)
+        .png({ compressionLevel })
```

Comment:
Use [toFormat(imageType, outputOptions)](http://sharp.dimens.io/en/stable/api-output/#toformat) instead to remove the if/else.